src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfiguredExceptionsStoreTests.cs

```diff
@@ -0,0 +1,299 @@
+        public async Task ConfiguredExceptionsStore_LimitOne_SingleRemainsWithInnerExceptions()
+        {
+            int ExpectedCount = 2;
+            int TopLevelLimit = 1;
+            ulong InnerExceptionId = 1;
+            ulong OuterExceptionId = 2;
+
+            ThresholdCallback callback = new(ExpectedCount);
+            await using ConfiguredExceptionsStore store = new ConfiguredExceptionsStore(TopLevelLimit, callback);
+
+            IExceptionsNameCache cache = CreateCache();
+
+            // Act
+            AddExceptionInstance(store, cache, InnerExceptionId, Array.Empty<ulong>());
+            AddExceptionInstance(store, cache, OuterExceptionId, new ulong[] { InnerExceptionId });
+
+            await callback.WaitForThresholdAsync(CommonTestTimeouts.GeneralTimeout);
+
+            // Assert
+            IReadOnlyList<IExceptionInstance> instances = store.GetSnapshot();
+            Assert.NotNull(instances);
+            Assert.Equal(ExpectedCount, instances.Count);
+
+            IExceptionInstance innerInstance = instances[0];
+            Assert.NotNull(innerInstance);
+            Assert.Equal(InnerExceptionId, innerInstance.Id);
+
+            IExceptionInstance outerInstance = instances[1];
+            Assert.NotNull(outerInstance);
+            Assert.Equal(OuterExceptionId, outerInstance.Id);
+        }
+
+        /// <summary>
+        /// Validates that adding a second top level exception to a store with a limit of one
+        /// will remove the older exception.
+        /// </summary>
+        [Fact]
+        public async Task ConfiguredExceptionsStore_LimitOne_LastRemains()
+        {
+            int ExpectedCount = 2;
+            int TopLevelLimit = 1;
+            ulong OuterException1Id = 1;
+            ulong OuterException2Id = 2;
+
+            ThresholdCallback callback = new(ExpectedCount);
+            await using ConfiguredExceptionsStore store = new ConfiguredExceptionsStore(TopLevelLimit, callback);
+
+            IExceptionsNameCache cache = CreateCache();
+
+            // Act
+            AddExceptionInstance(store, cache, OuterException1Id, Array.Empty<ulong>());
+            AddExceptionInstance(store, cache, OuterException2Id, Array.Empty<ulong>());
+
+            await callback.WaitForThresholdAsync(CommonTestTimeouts.GeneralTimeout);
+
+            // Assert
+            IReadOnlyList<IExceptionInstance> instances = store.GetSnapshot();
+            Assert.NotNull(instances);
+
+            IExceptionInstance outerInstance2 = Assert.Single(instances);
+            Assert.Equal(OuterException2Id, outerInstance2.Id);
+        }
+
+        /// <summary>
+        /// Validates that the last top level exception and its inner exceptions will remain
+        /// in the store with a limit of one.
+        /// </summary>
+        [Fact]
+        public async Task ConfiguredExceptionsStore_LimitOne_LastRemainsWithInnerExceptions()
+        {
+            int ExceptedInstanceCount = 4;
```

Comment:
nit: `ExpectedInstanceCount`

Reply:
I thought I fixed these, but apparently not.